jenkins/Jenkinsfile-blossom.premerge

```diff
@@ -345,11 +345,12 @@ void checkoutCode(String url, String sha) {
         poll: true,
         scm: [
             $class           : 'GitSCM', branches: [[name: sha]],
-            submoduleCfg     : [],
             userRemoteConfigs: [[
                                     credentialsId: 'github-token',
                                     url          : url,
-                                    refspec      : '+refs/pull/*/merge:refs/remotes/origin/pr/*']]
+                                    refspec      : '+refs/pull/*/merge:refs/remotes/origin/pr/*']],
+             extensions      : [[$class: 'CloneOption', shallow: true],
+                                [$class: 'SubmoduleOption', shallow: true, parentCredentials: true]]
```

Comment:
Can the premerge build work with a shallow clone?  I thought that was one of the few build pipelines that has the potential to checkout a different version than the default branch (e.g.: PR against branch-23.08 but default is branch-23.10 when it goes to get the merged head of PR to test).

Reply:
> 

Yes, it can clone with shallow as below:

```
Using shallow clone with depth 1
 > git fetch --tags --force --progress --depth=1 -- https://github.com/NVIDIA/spark-rapids.git 
Using shallow submodule update with depth 1
> git submodule init # timeout=10
> git submodule sync # timeout=10
```